### PR TITLE
Feature/sg 1060 yolo nas pose

### PR DESCRIPTION
**Pull request by BloodAxe**
This PR adds a final pieces of YoloNAS-Pose:
* Pretrained weights
* Integration tests
* Notebooks to run offline COCOEval
* Training recipes
* Docs
* Model zoo

### CODE REVIEW

File: YOLONAS-POSE.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# YOLO-NAS-POSE
+### A Next-Generation, Pose Estimation Foundational Model generated by Deci’s Neural Architecture Search Technology
+
+Deci is thrilled to announce the release of a new object detection model, YOLO-NAS-POSE - a successor of [YOLO-NAS](YOLONAS.md),
```
Comment (ofrimasad) [inline YOLONAS-POSE.md:4]: is it a successor? I would say a derivative

File: YOLONAS-POSE.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+| YOLO-NAS N       | 59.68 | 2.35 ms      |
+| YOLO-NAS S       | 64.15 | 3.29 ms      |
+| YOLO-NAS M       | 67.87 | 6.87 ms      |
+| YOLO-NAS L       | 68.24 | 8.86 ms      |
+
+AP numbers in table reported for COCO 2017 Val dataset and latency benchmarked for 640x640 images on Nvidia T4 GPU.
+No flip-TTA was used.
+
+Similarly to YOLO-NAS, YOLO-NAS-POSE architecture employs quantization-aware blocks and selective quantization for optimized performance. 
+In fact YOLO-NAS-POSE is a direct successor of YOLO-NAS and uses same backbone and neck as YOLO-NAS.
```
Comment (ofrimasad) [inline YOLONAS-POSE.md:28]: successor here again

File: documentation/source/models_export_pose.md [hunk 3]
```diff
@@ -0,0 +1,1738 @@
...
+
+```python
+np.set_printoptions(threshold=50, edgeitems=10)
+pred_poses, pred_poses.shape
+```
+
+
+
+
+    (array([[[[2.62617737e+02, 2.75986389e+02, 7.74692297e-01],
```
Comment (ofrimasad) [inline documentation/source/models_export_pose.md:345]: is that really necessary?

File: documentation/source/models_export_pose.md [hunk 4]
```diff
@@ -0,0 +1,1738 @@
...
+| 1          | Batch  | Yes                | No             | No             | Yes            |
+| >1         | Batch  | Yes                | No             | No             | Yes            |
+
+
+
+```python
+!pip install super-gradients==3.3.1
+```
+
+    ERROR: Could not find a version that satisfies the requirement super-gradients==3.3.1 (from versions: 1.3.0, 1.3.1, 1.4.0, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.7.5, 2.0.0, 2.0.1, 2.1.0, 2.2.0, 2.5.0, 2.6.0, 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.0.4, 3.0.5, 3.0.6, 3.0.7, 3.0.8, 3.0.9, 3.1.0, 3.1.1, 3.1.2, 3.1.3, 3.2.0, 3.2.1, 3.3.0)
```
Comment (ofrimasad) [inline documentation/source/models_export_pose.md:46]: What?

File: documentation/source/models_export_pose.md [hunk 5]
```diff
@@ -0,0 +1,1738 @@
...
+                                       providers=["CUDAExecutionProvider", "CPUExecutionProvider"])
+inputs = [o.name for o in session.get_inputs()]
+outputs = [o.name for o in session.get_outputs()]
+result = session.run(outputs, {inputs[0]: image_bchw})
+
+show_predictions_from_flat_format(image, result)
+```
+
+     25%|██████████████████████████████████████▊                                                                                                                    | 4/16 [00:12<00:37,  3.14s/it]
+    W1025 15:56:09.825534 23736 tensor_quantizer.py:174] Disable HistogramCalibrator
```
Comment (ofrimasad) [inline documentation/source/models_export_pose.md:1030]: can we drop this extremely long print?

**Review by ofrimasad — COMMENTED** [review]
Very impressive work @BloodAxe !
I have a few minor comments. Maybe we can clean some prints from the md files.

**Review by Louis-Dupont — APPROVED** [review]
Looks great! I left a few non blocking comments/questions

**Review by shaydeci — APPROVED** [review]
LGTM

File: Makefile [hunk 6]
```diff
@@ -32,6 +32,7 @@ sweeper_test:
 
 # Here you define a list of notebooks we want to execute and convert to markdown files
 NOTEBOOKS_TO_RUN := src/super_gradients/examples/model_export/models_export.ipynb
+NOTEBOOKS_TO_RUN += src/super_gradients/examples/model_export/models_export_pose.ipynb
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply Makefile:35]: There is none and I think it is for us to decide. 
IMO the right place for notebooks is under notebooks/ and not in the package source root (E.g notebooks a definitely not a part of wheel package right?)

File: documentation/source/models_export_pose.md (hunk 5 above)
Comment (BloodAxe) [reply documentation/source/models_export_pose.md:1030]: I think these warnings coming from pytorch_quantization package. 
There are few options but all looks dirty to me
1) Go and check whether we should address these warnings at all. (Long, out of the scope). And either fix or silence them inside  export() method
2) Globally set log level for ERROR in this notebook
3) `%%capture` in cell to just silence the output

File: YOLONAS-POSE.md (hunk 1 above)
Comment (BloodAxe) [reply YOLONAS-POSE.md]: Fixed in f4bc27730ea7141b7f07ea493d68bc8e61078776

File: YOLONAS-POSE.md (hunk 2 above)
Comment (BloodAxe) [reply YOLONAS-POSE.md]: Fixed in f4bc27730ea7141b7f07ea493d68bc8e61078776

File: documentation/source/models_export_pose.md (hunk 4 above)
Comment (BloodAxe) [reply documentation/source/models_export_pose.md]: Re-run notebook, fixed in c7c1be506c5c05665006d404a36ec7bd34ec2dd3

File: documentation/source/models_export_pose.md (hunk 5 above)
Comment (BloodAxe) [reply documentation/source/models_export_pose.md]: I re-run notebook and it went away (Probably thanks to the bug we have in loggers that causes all outputs to be muted in notebooks)

File: YOLONAS-POSE.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+#### COCO 2017
+
+* [super_gradients/recipes/coco2017_yolo_nas_pose_n.yaml](src/super_gradients/recipes/coco2017_yolo_nas_pose_n.yaml)
+* [super_gradients/recipes/coco2017_yolo_nas_pose_s.yaml](src/super_gradients/recipes/coco2017_yolo_nas_pose_s.yaml)
+* [super_gradients/recipes/coco2017_yolo_nas_pose_m.yaml](src/super_gradients/recipes/coco2017_yolo_nas_pose_m.yaml)
+* [super_gradients/recipes/coco2017_yolo_nas_pose_l.yaml](src/super_gradients/recipes/coco2017_yolo_nas_pose_l.yaml)
+
+
+## Additional resources
+
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply YOLONAS-POSE.md:65]: Fixed in 86e4073d878798f66fb65273981d1d321a945719

File: src/super_gradients/training/transforms/keypoints/keypoints_reverse_image_channels.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+from super_gradients.common.object_names import Processings
+from super_gradients.common.registry import register_transform
+from super_gradients.training.samples import PoseEstimationSample
+from .abstract_keypoints_transform import AbstractKeypointTransform
+
+
+@register_transform()
+class KeypointsReverseImageChannels(AbstractKeypointTransform):
+    """ """
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/training/transforms/keypoints/keypoints_reverse_image_channels.py]: Fixed in 20e6435080fe6c37b88a6f35ad13728009ffbad2

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply Makefile:35]: Yeah I tend to agree

**Review by ofrimasad — APPROVED** [review]
LGTM
